✨ add the css-utilities lib (#694)
Shopify/quilt: packages/react-css-utilities/README.md

# `@shopify/react-css-utilities`

[![Build Status](https://travis-ci.org/Shopify/quilt.svg?branch=master)](https://travis-ci.org/Shopify/quilt)
[![License: MIT](https://img.shields.io/badge/License-MIT-green.svg)](LICENSE.md) [![npm version](https://badge.fury.io/js/%40shopify%2Freact-css-utilities.svg)](https://badge.fury.io/js/%40shopify%2Freact-css-utilities.svg) [![npm bundle size (minified + gzip)](https://img.shields.io/bundlephobia/minzip/@shopify/react-css-utilities.svg)](https://img.shields.io/bundlephobia/minzip/@shopify/react-css-utilities.svg)

A set of react styling related utilities.

## Installation

```bash
$ yarn add @shopify/react-css-utilities
```

## Usage

> Both `classnames` and `variationName` utilities should ideally be use in combination with a `scss` styles sheet and [`css module`](https://github.com/css-modules/css-modules) setup to generate class name for react component.

Reviewer: A few comments:

- You jump right into this without even saying that these are two functions exported by the package
- `scss` here doesn't really matter
- `CSS` (uppercase)
- Again, I don't think React is relevant here.

Reviewer: Also it's `classNames` from the library

Reviewer: I think you should also link to the `classname` library, you don't really say what that function even does (removes falsy values, etc)

> Given a scss file for a simple Square

Reviewer: ```suggestion
Given this CSS file for a `Square` React component:
```

```scss
.Square {
  color: transparent;
}

.sizeSmall {
  height: 20px
  width: 20px
}

.sizeLarge {
  height: 44px
  width: 44px
}

.colorWhite {
  background-color: white;
}

.colorBlack {
  background-color: black;
}
```

> The following can be use to generate different class name from Props in a react component

Reviewer: ```suggestion
The following can be use to generate different class names for the component based on its props:
```

```tsx
interface Props {
  size: 'small' | 'large';
  color: 'white' | 'black';
}

function Square({size, color}: Props) {
  const className = classNames(
    styles.Square,
    styles[variationName('color', color)],
    styles[variationName('size', size)],
  );

  return <div className={className} />;
}
```

### Resulting className

> The final class name in the DOM will depend on your css module set up.

Reviewer: Don't bother saying this, the library has no impact on the built style sheet.
